# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/xf86-video-xgixp/hotfix.patch
# Copyright (C) 2022 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

From a84753dd495bcfd724406cae470e17d0c561d0c3 Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Tue, 25 Sep 2012 12:55:05 +0000
Subject: Remove mibstore.h

Signed-off-by: Adam Jackson <ajax@redhat.com>
---
diff --git a/src/xgi_driver.c b/src/xgi_driver.c
index 580a03a..8ce071b 100644
--- a/src/xgi_driver.c
+++ b/src/xgi_driver.c
@@ -53,8 +53,6 @@
 
 /* initialise a SW cursor */
 #include "mipointer.h"
-/* implementing backing store */
-#include "mibstore.h"
 
 /* colormap initialization */
 #include "micmap.h"
@@ -2041,10 +2039,6 @@ pScrn->pScreen = pScreen;
         xf86DrvMsg(pScrn->scrnIndex, X_ERROR, "FB Manager init failed \n");
     }
 
-    /* If backing store is to be supported (as is usually the case), initialise it. */
-    miInitializeBackingStore(pScreen);
-    XGIDebug(DBG_FUNCTION, "[DBG] Jong 06142006-After miInitializeBackingStore()\n");
-
     xf86SetBackingStore(pScreen);
     XGIDebug(DBG_FUNCTION, "[DBG] Jong 06142006-After xf86SetBackingStore()\n");
 
--
cgit v0.9.0.2-2-gbebe
--- xf86-video-xgixp-1.8.1/src/xgi_shadow.c.vanilla	2022-08-25 10:59:10.570230705 +0200
+++ xf86-video-xgixp-1.8.1/src/xgi_shadow.c	2022-08-25 10:59:22.773229978 +0200
@@ -86,7 +86,7 @@
 
 void XGIShadowUpdate (ScreenPtr pScreen, shadowBufPtr pBuf)
 {
-    RegionPtr       pRegion = &pBuf->damage;
+    RegionPtr       pRegion = &pBuf->pDamage;
     ScrnInfoPtr     pScrn;
     pScrn = xf86ScreenToScrn(pScreen);
 
--- xf86-video-xgixp-1.8.1/src/xgi_driver.c.vanilla	2022-08-25 10:55:53.160242472 +0200
+++ xf86-video-xgixp-1.8.1/src/xgi_driver.c	2022-08-25 10:56:04.716241783 +0200
@@ -1747,8 +1747,8 @@
     /* xf86DrvMsg(pScrn->scrnIndex, X_INFO, "++ Enter %s() %s:%d\n", __FUNCTION__, __FILE__, __LINE__); */
 #endif
 
-    pScreen->BlockHandler = pXGI->BlockHandler;
-    (*pScreen->BlockHandler)(BLOCKHANDLER_ARGS);
+    //pScreen->BlockHandler = pXGI->BlockHandler;
+    //(*pScreen->BlockHandler)(BLOCKHANDLER_ARGS);
     pScreen->BlockHandler = XGIBlockHandler;
 
 #ifdef XvExtension
